corehq/apps/accounting/models.py

```diff
@@ -857,7 +857,7 @@ class DefaultProductPlan(models.Model):
 
     class Meta(object):
         app_label = 'accounting'
-        unique_together = ('edition', 'is_trial', 'is_report_builder_enabled')
+        unique_together = ('edition', 'is_trial', 'is_report_builder_enabled', 'is_annual_plan')
```

Comment:
> Another way to say this is, some currently-deployed code would be broken and cause errors if the migrations in this PR were to be applied before the PR is deployed.
...
> Does this explanation help you form a better answer?

Yes, I think so. I believe this is closer to "include a migration that adds a new table and also include code that references that new table", in that the migration first alters a table, and then relies on that altered table state to add new software plans.

> I'm not sure how much data is in that table

Quite little - essentially one object for every uniquely named Software Plan, in this case a quick query returns a total of 12 `DefaultProductPlan`s on production, 10 on staging, and 15 (after running this migration) in my local environment.

I'm not really familiar with indices. Would creating the new index `CONCURRENTLY` first mean we're generating a new set of data with the change applied, waiting for that to complete, and then pointing the db to the new data? It seems like this might not be necessary given the small volume of data, but is it better practice to just go this route anyway?

> Finally, are there any potential uniqueness violations in the existing data?

There are not. `DefaultProductPlan` is only created in `ensure_plans`, which itself is only run in migrations. The `is_annual_plan` property was only recently added, with a default of `False` that is now applied to all instances existing prior to this PR, and because of the current constraint could not violate the new one.

Reply:
If there are so few (i.e., less than thousands) it will be no problem to do it this way, no need to do it `CONCURRENTLY`.